pkg/server/info_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+func createTestControllerInfo(ctx context.Context, k8sClient dynamic.Interface, name string) error {
+	controllerInfo := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": controllerInfoGVR.Group + "/" + controllerInfoGVR.Version,
+			"kind":       "AntreaControllerInfo",
+			"metadata": map[string]interface{}{
+				"name": name,
+			},
+			// an empty Spec is fine for the sake of the test
+			"spec": map[string]interface{}{},
+		},
+	}
+	_, err := k8sClient.Resource(controllerInfoGVR).Create(ctx, controllerInfo, metav1.CreateOptions{})
+	return err
+}
+
+func TestGetControllerInfo(t *testing.T) {
+	ctx := context.Background()
+	ts := newTestServer(t)
+	require.NoError(t, createTestControllerInfo(ctx, ts.k8sClient, "antrea-controller"), "failed top create test ControllerInfo")
```

Comment:
```suggestion
	require.NoError(t, createTestControllerInfo(ctx, ts.k8sClient, "antrea-controller"), "failed to create test ControllerInfo")
```
Same for several places in this file